server/src/index.ts

```diff
@@ -27,6 +28,7 @@ export function startApp(): Koa {
     // Enable bodyParser with default options
     app.use(bodyParser());
 
+    app.use(authValidator);
```

Comment:
this should go in after the unprotected routes, before the protected routes, such that if the user can not be authenticated, a 401 is returned before they're able to get to the protected routes. All middleware which should be run on every request should of course be put before this.

Reply:
We can't just blanket place protected routes ahead of unprotected routes and stop processing if a user is unauthenticated because the endpoint that serves the html MUST come at the end. 

As for moving the isAuthenticated middleware out of the router, I can look into placing it into the index but I'm not sure if it's worth the time investment.